io/include/detray/io/csv/track_parameters.hpp

```diff
@@ -134,7 +134,9 @@ inline void write_free_track_params(
         for (const auto &[charge, track_param] : track_params) {
 
             const auto &glob_pos = track_param.pos();
-            const auto &p = track_param.mom(charge);
+            // Momentum may not be retrievable for straight-line tracks
```

Comment:
Isn't the straight line track supposed to have zero curvature in the trajectory.. but is it necessarily set qop to zero and momentum undefined? For such setup, we can not apply momentum-dependent functions to tracks, e.g. material interaction

Reply:
I am not sure I understand the comment. What do you want me to do here?

This is a function to write track parameters to csv and that includes the momentum. For rays ```qop``` was set to zero in a previous PR and hence I have to make a distinction here.

Reply:
What I am saying is that you don't have to change this in this PR because straight line track does not mean it has no momentum.

For zero qop in the earlier PR, we might need to rethink this as well.

Reply:
The way it is currently implemented, this triggers an assertion in the track parameters, because we are dividing by qop (which can be zero) to obtain the momentum. In order for the straight line navigation validation to work, the track momentum needs to be written to file and read in at later step. So I need to do something to fix it here, even if it remains a hotfix. I had ```qop``` initially set to invalid for rays, which is also not correct.

Reply:
OK I see. I think that happens because the csv edm (q,p) is different from core edm (q/p, n).

Reply:
Indeed, but I need the concrete charge value in order to infer the momentum and call the constructor of the track parameters, so qop is not enough here, I think? Also, this might be helpful to set the particle hypothesis somewhere...